### PR TITLE
Spyglass job history page error: correctly detecting http response code

### PR DESCRIPTION
**Pull request by chaodaiG**

### CODE REVIEW

File: prow/cmd/deck/main_test.go [hunk 1]
```diff
@@ -1189,6 +1189,14 @@ func TestHttpStatusForError(t *testing.T) {
 			},
 			expectedStatus: http.StatusGone,
 		},
+		{
```
Comment (chaodaiG) [inline prow/cmd/deck/main_test.go:1192]: The current way of retrieving http response code always return 5XX for any wrapped errors, see this test case

**Comment by chaodaiG** [comment]
Fixes: https://github.com/kubernetes/test-infra/issues/21980

File: prow/cmd/deck/main_test.go (hunk 1 above)
Comment (alvaroaleman) [reply prow/cmd/deck/main_test.go:1192]: Shouldn't we fix that then?

Comment (chaodaiG) [reply prow/cmd/deck/main_test.go:1192]: Agree that's the best approach, thought about it for a few minutes but didn't have a good way. Will update here if I can find a better way. Also let me know if you have an idea

Comment (alvaroaleman) [reply prow/cmd/deck/main_test.go:1192]:
```
diff --git a/prow/cmd/deck/main.go b/prow/cmd/deck/main.go
index dec42e1b0e..94ad16678c 100644
--- a/prow/cmd/deck/main.go
+++ b/prow/cmd/deck/main.go
@@ -1651,7 +1651,8 @@ type httpError struct {
 }
 
 func httpStatusForError(e error) int {
-       if httpErr, ok := e.(httpError); ok {
+       var httpErr httpError
+       if ok := errors.As(e, &httpErr); ok {
                return httpErr.statusCode
        }
        return http.StatusInternalServerErro
```

Comment (chaodaiG) [reply prow/cmd/deck/main_test.go:1192]: Awesome, good to learn! I'll give it a shot

Comment (chaodaiG) [reply prow/cmd/deck/main_test.go:1192]: Done. And btw replaced a few `%v` to `%w`

Comment (chaodaiG) [reply prow/cmd/deck/main_test.go:1192]: Also updated both PR title and commit msg

Comment (alvaroaleman) [reply prow/cmd/deck/main_test.go:1192]: Thanks. As an aside, we have verification in our downstream repo to enforce error wrapping: https://github.com/openshift/ci-tools/blob/9f710b04bc1319b5374ea7ef26a8dd1e8dd97f8b/hack/lint.sh#L24

And also something to fix it up automatically that works okayish: https://github.com/openshift/ci-tools/blob/9f710b04bc1319b5374ea7ef26a8dd1e8dd97f8b/hack/fixup-error-wrapping.sh#L10

However, I do not know how to integrate new stuff with bazel :upside_down_face:

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**